File: VB/Default MVVM/FormatConditionsProvider/FormatCondition.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Windows.Media
Imports DevExpress.Data.Filtering
Imports System.ComponentModel

Namespace Default_MVVM
	Public Class FormatCondition
		Implements INotifyPropertyChanged

		' Fields...
		Private _FieldName As String
		Private _Criteria As CriteriaOperator
		Private _Appearance As Appearance

		Public Property Appearance() As Appearance
			Get
				If _Appearance Is Nothing Then
					_Appearance = New Appearance()
				End If
				Return _Appearance
			End Get
			Set(ByVal value As Appearance)
                If Object.Equals(_Appearance, value) Then
                    _Appearance = value
                    OnPropertyChanged("Appearance")
                End If
			End Set
		End Property

		Public Property Criteria() As CriteriaOperator
			Get
				Return _Criteria
			End Get
			Set(ByVal value As CriteriaOperator)
				If (Not Object.Equals(_Criteria, value)) Then
					_Criteria = value
					OnPropertyChanged("Criteria")
				End If
			End Set
		End Property

		Public Property FieldName() As String
			Get
				Return _FieldName
			End Get
			Set(ByVal value As String)
				If _FieldName <> value Then
					_FieldName = value
					OnPropertyChanged("FieldName")
				End If
			End Set
		End Property

		Public Overrides Function ToString() As String
			Return String.Format("Condition FieldName = {0}, Criteria = {1}", FieldName, Criteria)
		End Function

		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Private Sub OnPropertyChanged(ByVal propertyName As String)
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
	End Class
End Namespace

File: CS/Default MVVM/FormatConditionsProvider/FormatCondition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Media;
using DevExpress.Data.Filtering;
using System.ComponentModel;

namespace Default_MVVM
{
    public class FormatCondition : INotifyPropertyChanged
    {

        // Fields...
        private string _FieldName;
        private CriteriaOperator _Criteria;
        private Appearance _Appearance;

        public Appearance Appearance
        {
            get
            {
                if (_Appearance == null)
                    _Appearance = new Appearance();
                return _Appearance;
            }
            set
            {
                if (_Appearance != value)
                {
                    _Appearance = value;
                    OnPropertyChanged("Appearance");
                }
            }
        }

        public CriteriaOperator Criteria
        {
            get { return _Criteria; }
            set
            {
                if (!object.Equals(_Criteria, value))
                {
                    _Criteria = value;
                    OnPropertyChanged("Criteria");
                }
            }
        }

        public string FieldName
        {
            get { return _FieldName; }
            set
            {
                if (_FieldName != value)
                {
                    _FieldName = value;
                    OnPropertyChanged("FieldName");
                }
            }
        }

        public override string ToString()
        {
            return string.Format("Condition FieldName = {0}, Criteria = {1}", FieldName, Criteria);
        }

        public event PropertyChangedEventHandler PropertyChanged;

        void OnPropertyChanged(string propertyName)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}
